components/generic_modal.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import './generic_modal.scss';
+
+type Props = {
+    onHide: () => void;
+    modalHeaderText: React.ReactNode;
+    show?: boolean;
+    handleCancel?: () => void;
+    handleConfirm?: () => void;
+    confirmButtonText?: React.ReactNode;
+    confirmButtonClassName?: string;
+    cancelButtonText?: React.ReactNode;
+    isConfirmDisabled?: boolean;
+};
+
+type State = {
+    show: boolean;
+}
+
+export default class GenericModal extends React.PureComponent<Props, State> {
```

Comment:
What is the benefit of a class vs. functional component here?

Reply:
Maybe I'm not understanding functional components properly. but this component does have some internal logic and state that need to be defined, so I didn't think that was possible with an FC.

Reply:
My idea was to try move towards FC with the use of React hooks. You can achieve all that using this unless you are exposing the instance methods via refs (we should also try avoiding that btw :P). But I won't block this PR.

Reply:
Do you feel like it's worth reworking this component to do that? I'm not sure about the effort since I haven't worked with hooks yet, but I guess I need to try it out at some point.

Reply:
No you are right, I won't block this PR but we can discuss in the Hangouts how we wan't to proceed with hooks and FC.